packages/graphiql-toolkit/src/types.ts

```diff
@@ -35,20 +35,23 @@ export type FetcherOpts = {
   documentAST?: DocumentNode;
 };
 
-export type FetcherResult =
+export type IncrementalDeliveryResult = {
+  data?: any;
+  errors?: any[];
+  path?: [string, number];
+  hasNext: boolean;
+};
+
+export type FetcherResultObject =
   | {
       data: IntrospectionQuery;
       errors?: Array<any>;
     }
-  | string
   | { data?: any; errors?: Array<any>; hasNext?: boolean }
   // for IncrementalDelivery
-  | Array<{
-      data?: any;
-      errors?: any[];
-      path?: [string, number];
-      hasNext: boolean;
-    }>;
+  | IncrementalDeliveryResult;
+
+export type FetcherResult = FetcherResultObject | string;
```

Comment:
I don't think we need to compose them into distinct types to use seperately? `FetcherResult` infers the right type if you use it properly! were there issues with that for you?

Reply:
I think what I wanted here was to have a concrete type, as in to target a specific element in that union when i did my type guard.

But ya, think we can ditch that now. with the `hasNext?: boolean` thingo.